core/modules/replica/IngestServerConnection.cc

```diff
@@ -191,6 +195,13 @@ void IngestServerConnection::_handshakeReceived(boost::system::error_code const&
     _isOverlap       = request.is_overlap();
     _columnSeparator = request.column_separator() == ProtocolIngestHandshakeRequest::COMMA ? ',' : '\t';
 
+    // Check if the client is authorized for the operation
+
+    if (request.auth_key() != _authKey) {
```

Comment:
But this isn't separated out into a function...

Reply:
Good point!